tgui/packages/tgui/interfaces/SlotMachine.tsx

```diff
@@ -0,0 +1,151 @@
+import { useBackend } from '../backend';
+import { Button, Icon, Section } from '../components';
+import { Window } from '../layouts';
+
+type IconInfo = {
+  value: number;
+  colour: string;
+  icon_name: string;
+};
+
+type BackendData = {
+  icons: IconInfo[];
+  state: any[];
+  balance: number;
+  working: boolean;
+  money: number;
+  cost: number;
+  plays: number;
+  jackpots: number;
+  jackpot: number;
+};
+
+type SlotsTileProps = {
+  icon: string;
+  color?: string;
+  background?: string;
+};
+
+type SlotsReelProps = {
+  reel: IconInfo[];
+};
+
+const randomChoice = (arr: any[]) => {
+  return arr[Math.floor(Math.random() * arr.length)];
+};
+
+const pluralS = (amount: number) => {
+  return amount === 1 ? '' : 's';
+};
+
+const SlotsReel = (props: SlotsReelProps) => {
+  const { reel } = props;
+  return (
+    <div
+      style={{
+        display: 'inline-flex',
+        flexDirection: 'column',
+      }}
+    >
+      {reel.map((slot, i) => (
+        <SlotsTile key={i} icon={slot.icon_name} color={slot.colour} />
+      ))}
+    </div>
+  );
+};
+
+const SlotsTile = (props: SlotsTileProps) => {
+  return (
+    <div
+      style={{
+        textAlign: 'center',
+        padding: '1rem',
+        margin: '0.5rem',
+        display: 'inline-block',
+        width: '5rem',
+        background: props.background || 'rgba(62, 97, 137, 1)',
+      }}
+    >
+      <Icon className={`color-${props.color}`} size={2.5} name={props.icon} />
+    </div>
+  );
+};
+
+export const SlotMachine = (props) => {
+  const { act, data } = useBackend<BackendData>();
+  // icons: The list of possible icons, including colour and name
+  // backendState: the current state of the slots according to the backend
+  const {
+    plays,
+    jackpots,
+    money,
+    cost,
+    state,
+    balance,
+    jackpot,
+    working: rolling,
+  } = data;
+
+  return (
+    <Window>
+      <Section style={{ justifyContent: 'center', textAlign: 'center' }}>
+        <h1>Slots!</h1>
```

Comment:
section has a title prop, u can use that to set the header instead of the bottom line